docs/source/deployment/CORS_Headers.rst

```diff
@@ -0,0 +1,173 @@
+   Access-Control-Allow-Origin: "http://motech-ui.example | http://other.domain" (allow access from two mentioned origins)
+
+Access-Control-Allow-Methods
+----------------------------
+
+Comma-delimited list of the supported HTTP methods (`GET`, `HEAD`, `POST`, `CONNECT`, `TRACE`, `TRACK`,  `DELETE`, `OPTIONS`, `PUT`). Examples:
+
+.. code-block::
+
+   Access-Control-Allow-Methods: "GET" (only GET method is allowed in request)
+   Access-Control-Allow-Methods: "POST, GET" (POST and GET are allowed in request)
+
+Access-Control-Allow-Headers
+----------------------------
+
+This header specifies which not simple HTTP headers can be used in a request to MOTECH-CORE. Example :
+
+.. code-block::
+
+   Access-Control-Allow-Headers: "Content-Type,X-Requested-With,Accept,Authorization,Origin,Access-Control-Request-Method,Access-Control-Request-Headers"
+
+Access-Control-Expose-Headers
+-----------------------------
+
+During a CORS request, client(MOTECH-UI) can only access simple response headers. Simple response headers are defined as follows:
+    - Cache-Control
+    - Content-Language
+    - Content-Type
+    - Expires
+    - Last-Modified
+    - Pragma
+
+If you want clients to be able to access other headers, you have to use the Access-Control-Expose-Headers header. The value of this header is a comma-delimited list of response headers you want to expose to the client. Example :
+
+.. code-block::
+
+   Access-Control-Expose-Headers: "Access-Control-Allow-Origin,Access-Control-Allow-Credentials" (client has access to values of mentioned headers)
+
+Apache Config
+=============
+
+Apache includes support for CORS. To enable CORS support we have to running the following command which enable necessary for CORS mod headers :
+
+	a2enmod headers
+
+Now we can add the CORS headers into server configuration (usually /etc/apache*/apache*.conf). Inside a tag '<directory **pathToMotech**>', we set up CORS headers which we want to add into server configuration.
+
+Here is a config example "allow-all-origins":
+
+.. code-block::
+
+    <directory /exampleMotechDir/>
+        Header always set Access-Control-Allow-Origin "*"
+    </directory>
+
+Here is other config example (let's assume that http://motech-ui.example is an url to MOTECH-UI application):
+
+.. code-block::
+
+    <directory /exampleMotechDir/>
+        Header always set Access-Control-Allow-Origin "http://motech-ui.example"
+        Header always set Access-Control-Allow-Headers "Content-Type,X-Requested-With,Accept,Authorization,Origin,Access-Control-Request-Method,Access-Control-Request-Headers"
+        Header always set Access-Control-Expose-Headers "Access-Control-Allow-Origin,Access-Control-Allow-Credentials"
+        Header always set Access-Control-Allow-Methods "POST, GET, OPTIONS, DELETE, PUT"
+    </directory>
+
+If changes are made while the Apache server was running, you should restart apache.
+
+Tomcat Config
```

Comment:
If I proxy to Tomcat from Apache, I assume I don't need this? This is only if expose Tomcat directly to the web?